src/dali/plugin/input/rest/kraken.py

```diff
@@ -0,0 +1,454 @@
+                        )
+                    )
+                else:
+                    crypto_out_no_fee: str = str(amount)
+                    crypto_out_with_fee: str = str(amount + RP2Decimal(record[_FEE]))
+                    fiat_out_no_fee: str = str(RP2Decimal(trade_history[record[_REFID]][_COST]) - RP2Decimal(
+                        trade_history[record[_REFID]][_FEE]))
+
+                    result.append(
+                        OutTransaction(
+                            plugin=self.__PLUGIN_NAME,
+                            unique_id=unique_id,
+                            raw_data=raw_data,
+                            timestamp=timestamp_value,
+                            asset=asset_base,
+                            exchange=self.__EXCHANGE_NAME,
+                            holder=self.account_holder,
+                            transaction_type=transaction_type,
+                            spot_price=spot_price,
+                            crypto_out_no_fee=crypto_out_no_fee,
+                            crypto_fee=crypto_fee,
+                            crypto_out_with_fee=crypto_out_with_fee,
+                            fiat_out_no_fee=fiat_out_no_fee,
+                            fiat_fee=fiat_fee,
+                            notes=key,
+                        )
+                    )
+            elif record[_TYPE] == _MARGIN or record[_TYPE] == _ROLLOVER:
+                self.__logger.debug("Trade history record: %s", trade_history[record[_REFID]])
+
+                spot_price = Keyword.UNKNOWN.value
+                crypto_out_no_fee = str(amount)
+                crypto_out_with_fee = str(amount + RP2Decimal(record[_FEE]))
+                fiat_out_no_fee = str(RP2Decimal(trade_history[record[_REFID]][_COST]) - RP2Decimal(
+                    trade_history[record[_REFID]][_FEE]))
+
+                result.append(
+                    OutTransaction(
+                        plugin=self.__PLUGIN_NAME,
+                        unique_id=Keyword.UNKNOWN.value,
+                        raw_data=raw_data,
+                        timestamp=timestamp_value,
+                        asset=asset_base,
+                        exchange=self.__EXCHANGE_NAME,
+                        holder=self.account_holder,
+                        transaction_type=Keyword.SELL.value,
+                        spot_price=spot_price,
+                        crypto_out_no_fee=crypto_out_no_fee,
+                        crypto_fee=crypto_fee,
+                        crypto_out_with_fee=crypto_out_with_fee,
+                        fiat_out_no_fee=fiat_out_no_fee,
+                        fiat_fee=fiat_fee,
+                        notes=key,
+                    )
+                )
+            elif record[_TYPE] == _TRANSFER:
+                spot_price = Keyword.UNKNOWN.value
+                crypto_in = str(amount)
+                fiat_in_no_fee = '0'
+                fiat_in_with_fee = '0'
```

Comment:
I suggest removing the fiat value, unless you are initializing them to a positive value. If you leave them at 0 and the crypto value is > 0 RP2 will issue a warning. Instead if you skip them altogether (they are optional anyway), RP2 will fill them with the converted value (spot_price * crypto).